lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs

```diff
@@ -742,46 +741,26 @@ balanceTransactionInner
     = do
     (balance0, minfee0, _) <- balanceAfterSettingMinFee partialTx
 
-    (extraInputs, extraCollateral', extraOutputs, s') <- do
-
-        -- NOTE: It is not possible to know the script execution cost in
-        -- advance because it actually depends on the final transaction. Inputs
-        -- selected as part of the fee balancing might have an influence on the
-        -- execution cost.
-        -- However, they are bounded so it is possible to balance the
-        -- transaction considering only the maximum cost, and only after, try to
-        -- adjust the change and ExUnits of each redeemer to something more
-        -- sensible than the max execution cost.
-
-        let
-            transform
-                :: Selection
-                -> ( [(W.TxIn, W.TxOut)]
-                   , [(W.TxIn, W.TxOut)]
-                   , [W.TxOut]
-                   , changeState
-                   )
-            transform sel =
-                let (sel', s') = assignChangeAddresses genChange sel s
-                    inputs = F.toList (sel' ^. #inputs)
-                in  ( inputs
-                    , sel' ^. #collateral
-                    , sel' ^. #change
-                    , s'
-                    )
-
-        let mSel = selectAssets
-                pp
-                utxoAssumptions
-                (F.toList $ partialTx ^. bodyTxL . outputsTxBodyL)
-                redeemers
-                utxoSelection
-                balance0
-                (Convert.toWalletCoin minfee0)
-                genChange
-                selectionStrategy
-
-        transform <$> mSel
+    -- NOTE: It is not possible to know the script execution cost in
+    -- advance because it actually depends on the final transaction. Inputs
+    -- selected as part of the fee balancing might have an influence on the
+    -- execution cost.
+    -- However, they are bounded so it is possible to balance the
+    -- transaction considering only the maximum cost, and only after, try to
+    -- adjust the change and ExUnits of each redeemer to something more
+    -- sensible than the max execution cost.
+    (Selection extraInputs extraCollateral extraOutputs extraInputScripts, s')
```

Comment:
When pattern matching on this record type, I would propose to use named field puns here, if possible.

Something like this:
```hs
Selection {extraInputs, extraCollateral} <- RHS
```
One reason is that some of the fields share the same type. For example:
```hs
extraInputs     :: [W.TxIn]
extraCollateral :: [W.TxIn]
``` 
With positional matching, it is rather easy to inadvertently switch the arguments.

If the resultant line is too long, then perhaps first we could pattern match on the pair, and then on a subsequent line create a `let` binding that pattern matches on the individual fields.

Reply:
👍 I had it in a past version, but lost when rebasing.